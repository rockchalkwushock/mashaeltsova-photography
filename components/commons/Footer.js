import { styles } from '../../lib'

const Footer = ({ children }) => (
  <footer>
    {children}
    <style jsx>{`
      footer {
        background-color: ${styles.colors.peach};
        display: flex;
        flex-direction: column;
        padding: 1em;
      }
      @media (min-width: 768px) {
        footer {
          align-items: center;
          flex-direction: row;
        }
      }
    `}</style>
  </footer>
)

export default Footer
