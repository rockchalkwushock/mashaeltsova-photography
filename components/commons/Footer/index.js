/**
 * REVIEW
 * FIXME
 * I will need to pass props on this like `intl`
 */
const Footer = ({ children }) => (
  <footer>
    {children}
    <style>{`
    footer {
      background-color: lightcoral;
      grid-column: span 7;
      height: 50px;
    }
  `}</style>
  </footer>
)

export default Footer
