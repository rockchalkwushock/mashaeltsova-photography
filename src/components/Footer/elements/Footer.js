import styled from 'styled-components'

const Footer = styled.footer`
  background-color: #333;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-top: 2em;
  text-align: center;
  width: 100vw;
  @media (min-width: ${({ theme }) => theme.sizes.md}) {
    align-items: center;
    flex-direction: row;
    justify-content: space-between;
    padding: 2em 1em 1em 1em;
  }
`

Footer.displayName = 'Footer'

export default Footer
