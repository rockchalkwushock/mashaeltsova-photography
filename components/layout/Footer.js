import { Container, Icon } from '../commons'
import { styles } from '../../lib'

const Footer = () => (
  <footer>
    <Container className="social">
      <Icon
        icon="facebook"
        size="2x"
        url="https://www.facebook.com/masha.eltsova"
      />
      <Icon
        icon="instagram"
        size="2x"
        url="https://www.instagram.com/mashaeltcovaphotography"
      />
      <Icon icon="vk" size="2x" url="https://www.vk.com/club65938200" />
      <Icon icon="twitter" size="2x" url="https://twitter.com/MASHAELTSOVA" />
    </Container>
    <Container className="copyright">
      <Icon icon="copyright" size="lg" url="https://codybrunner.rocks" />
    </Container>
    <style jsx>{`
      footer {
        background-color: ${styles.colors.peach};
        display: -webkit-box;
        display: -ms-flexbox;
        display: flex;
        -webkit-box-orient: vertical;
        -webkit-box-direction: normal;
        -ms-flex-direction: column;
        flex-direction: column;
        padding: 1em;
      }
      @media (min-width: 768px) {
        footer {
          -webkit-box-align: center;
          -ms-flex-align: center;
          align-items: center;
          -webkit-box-orient: horizontal;
          -webkit-box-direction: normal;
          -ms-flex-direction: row;
          flex-direction: row;
        }
      }
    `}</style>
  </footer>
)

export default Footer
