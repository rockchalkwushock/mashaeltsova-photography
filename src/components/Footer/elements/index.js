import Footer from './Footer'
import FooterContainer from './FooterContainer'

export { Footer, FooterContainer }
