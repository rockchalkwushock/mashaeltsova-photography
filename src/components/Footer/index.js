import React from 'react'

import { Footer, FooterContainer } from './elements'

const AppFooter = () => (
  <Footer>
    <FooterContainer left>
      <h2>Left</h2>
    </FooterContainer>
    <FooterContainer>
      <h2>Right</h2>
    </FooterContainer>
  </Footer>
)

export default AppFooter
