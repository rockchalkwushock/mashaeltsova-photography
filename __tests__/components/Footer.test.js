import React from 'react'
import { render } from 'enzyme'

import Footer from '../../components/layout/Footer'

describe('Component: Footer', () => {
  test('Renders without exploding', () => {
    const tree = render(<Footer />)
    expect(tree).toMatchSnapshot()
  })
})
