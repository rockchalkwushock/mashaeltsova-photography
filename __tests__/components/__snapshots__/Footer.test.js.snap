// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Footer Renders without exploding 1`] = `
<footer
  class="jsx-747610455"
>
  <div
    class="jsx-590062731 social"
  >
    <a
      href="https://www.facebook.com/masha.eltsova"
    >
      <span
        aria-hidden="true"
        class="fa fa-facebook fa-2x"
      />
    </a>
    <a
      href="https://www.instagram.com/mashaeltcovaphotography"
    >
      <span
        aria-hidden="true"
        class="fa fa-instagram fa-2x"
      />
    </a>
    <a
      href="https://www.vk.com/club65938200"
    >
      <span
        aria-hidden="true"
        class="fa fa-vk fa-2x"
      />
    </a>
    <a
      href="https://twitter.com/MASHAELTSOVA"
    >
      <span
        aria-hidden="true"
        class="fa fa-twitter fa-2x"
      />
    </a>
  </div>
  <div
    class="jsx-590062731 copyright"
  >
    <div
      class="jsx-1032329529"
    >
      <a
        class="jsx-1032329529"
        href="https://codybrunner.rocks"
      >
        <span
          class="jsx-1032329529"
        >
          Designed by Cody Brunner
        </span>
      </a>
      <span
        class="jsx-1032329529"
      >
        All images 
        <span
          aria-hidden="true"
          class="fa fa-copyright fa-lg"
        />
         2017 Masha Eltsova Photography
      </span>
    </div>
  </div>
</footer>
`;
